include/mfmg/dealii/lanczos.templates.hpp

```diff
@@ -263,34 +263,28 @@ Lanczos<OperatorType, VectorType>::details_calc_tridiag_epairs(
   if (n < num_requested)
     return std::make_tuple(evals, evecs);
 
-  // Allocate storage
-  std::vector<double> matrix(n * n, 0.);
-
-  // Copy diagonals of the tridiagonal matrix into the full matrix
-  matrix[0] = t_maindiag[0];
-  for (int i = 1; i < n; ++i)
-  {
-    matrix[i + n * i] = t_maindiag[i];
-    matrix[i - 1 + n * i] = t_offdiag[i - 1];
-  }
+  // dstyev destroys storage
+  evals = main_diagonal;
+  std::vector<double> sub_diagonal_aux = sub_diagonal;
+  std::vector<double> evecs_aux(n * n);
 
   // Eigenvalues are returned in ascending order
-  evals.resize(n);
-  const lapack_int info = LAPACKE_dsyev(LAPACK_COL_MAJOR, 'V', 'U', n,
-                                        matrix.data(), n, evals.data());
-  ASSERT(!info, "Call to LAPACKE_dsyev failed.");
+  const lapack_int info =
+      LAPACKE_dstev(LAPACK_COL_MAJOR, 'V', n, evals.data(),
+                    sub_diagonal_aux.data(), evecs_aux.data(), n);
+  ASSERT(!info, "Call to LAPACKE_dstev failed.");
```

Comment:
I remember you changing from DGEEV (a.k.a "simple driver for eigenvalues/vectors" for Nonsymmetric Eigenproblems) to DSYEV ("simple driver" for Symmetric Eigenproblems) in 495ac3d and I will admit I did not bother looking at the documentation at the time.  Now you are changing it to DSTEV ("simple driver (tridiagonal matrix)" for Symmetric Eigenproblems) which, now that I have looked into LAPACK's user guide, looks like the right thing to do.  Please add a comment and make sure to add a reference to the documentation on netlib.org.

Reply:
Don't see a need to, it's a standard easily searchable name.

Reply:
Don't make me change my review from "Approve" to "Request changes" :p

Reply:
Addressed.